test/AlgorithmStub.hpp

```diff
@@ -0,0 +1,23 @@
+#include "Algorithm.hpp"
+
+class AlgorithmStubA : public Algorithm
+{
+public:
+    AlgorithmStubA() {}
```

Comment:
no need to explicitly write default constructor here :) 
Less code = less places for mistakes = less maintenance = better code

Reply:
done